src/app/transactions/MempoolFeePieChart.tsx

```diff
@@ -0,0 +1,139 @@
+import { useColorModeValue } from '@chakra-ui/react';
+import { Cell, Pie, PieChart } from 'recharts';
+
+const pieChartWidth = 215;
+const pieChartHeight = 200;
+
+const renderCustomizedLabel = ({
+  percent,
+  cx,
+  cy,
+  midAngle,
+  outerRadius,
+}: {
+  percent: number;
+  x: number;
+  y: number;
+  cx: number;
+  cy: number;
+  midAngle: number;
+  outerRadius: number;
+}) => {
+  if (!percent) return null; // Don't show zeros
+  const RADIAN = Math.PI / 180;
+  const radius = outerRadius + 25; // Increase radius to move label further out
+  const newX = cx + radius * Math.cos(-midAngle * RADIAN);
+  const newY = cy + radius * Math.sin(-midAngle * RADIAN);
+  return (
+    <text
+      x={newX}
+      y={newY}
+      fill="#737680"
+      fontSize="12px"
+      textAnchor="middle"
+      dominantBaseline="central"
+    >
+      {`${(percent * 100).toFixed(0)}%`}
+    </text>
+  );
+};
+
+const renderCenterCustomizedLabel = ({
+  label,
+  cx,
+  cy,
+  color,
+}: {
+  label: string;
+  cx: number;
+  cy: number;
+  color: string;
+}) => {
+  console.log('color', color);
+  // const textColor = useColorModeValue('text', 'white');
+  return (
+    <text
+      x={cx}
+      y={cy}
+      // color="text" // TODO: change for light and dark mode
+      textAnchor="middle"
+      dominantBaseline="central"
+      style={{
+        fill: color,
+        fontSize: '20px',
+        fontStyle: 'normal',
+        fontWeight: 500,
+        lineHeight: '24px',
+        letterSpacing: '-0.2px',
+      }}
+    >
+      {label}
+    </text>
+  );
+};
+
+export function getTxTypePieChartColor(txType: string) {
+  switch (txType) {
+    case 'token_transfer':
+      return '#9985FF';
```

Comment:
We aim to use colors exclusively from theme color palette.

Reply:
These were the colors that were on the figma design. I can add these colors to the theme color palette

Reply:
@eugeniadigon can you let me know where these colors fit in our colors theme please? See https://github.com/hirosystems/explorer/blob/main/src/ui/theme/colors.ts

Reply:
@eugeniadigon pinging on this

Reply:
These are new colors outside of the color palette we already have, which only covers slate (neutrals palette), purple (stacks color palette), and error/warning/success (red, orange, green). For this, I needed other colors to refer to transactions types, otherwise it would've looked like 'more of the same' (using purple or slate) or error/warnings (using red/orange/green). What can I do to help here? Add these to the Figma UI theme?

Reply:
I'd add any value to the Figma UI theme. Once we start using hard corded values we'll have the risk of loose consistency, specially if these colors might end up being used somewhere else

Reply:
Sounds good, [added here](https://www.figma.com/file/6JPmaDsHZ2fHfOyNY7sIZY/Stacks-Explorer-UI-Theme?type=design&node-id=1%3A203&mode=design&t=IHcN68zqiIH8tUlz-1)